packit/utils/obs_helper.py

```diff
@@ -0,0 +1,454 @@
+
+    try:
+        with open(changes_filename, "w") as changes_file:
+            changes_file.write(changes_file_content)
+    except OSError as err:
+        logger.warn(f"Error writing {changes_filename} file: {err}")
+
+
+def create_package(project_name: str, package_name: str) -> None:
+    """Creates the package with the name ``package_name`` in the project
+    ``project_name`` on OBS. No sources are uploaded in the process.
+
+    """
+    (root := ET.Element("package")).attrib["name"] = package_name
+    root.attrib["project"] = project_name
+
+    (title := ET.Element("title")).text = f"The {package_name} package"
+    descr = ET.Element("description")
+
+    root.append(title)
+    root.append(descr)
+
+    package_url = core.makeurl(
+        _API_URL,
+        ["source", project_name, package_name, "_meta"],
+    )
+    metafile = core.metafile(package_url, ET.tostring(root))
+    metafile.sync()
+
+
+def create_obs_project(
+    project: str,
+    targets: str,
+    owner: Optional[str],
+    package_config: PackageConfig,
+    description: Optional[str],
+):
+    """
+    Creates a new OBS project and its associated package, ensuring only a single
+    package is included per call.
+
+    Args:
+        project (str, optional): The desired name for the OBS project. Defaults to None.
+        targets (str): Comma-separated list of build targets for the project. Defaults to "".
+        owner (Optional[str], optional): The owner of the project. Defaults to None.
+        package_config (PackageConfig): The config for the package to be included in the project.
+        description (Optional[str], optional): A description for the project. Defaults to None.
+
+    Returns:
+        Tuple[str, str]: A tuple containing the created project name and associated package name.
+
+    Raises:
+        ValueError: If the provided package_config contains multiple packages.
+    """
+    conf.get_config()
+    owner = owner or conf.config["api_host_options"][_API_URL]["user"]
+    project_name = project or f"home:{owner}:packit"
+
+    targets_list = targets.split(",")
+    for target in targets_list:
+        if target in DEPRECATED_TARGET_MAP:
+            logger.warning(
+                f"Target '{target}' is deprecated. "
+                f"Please use '{DEPRECATED_TARGET_MAP[target]}' instead.",
+            )
+
+    project_metadata = targets_to_project_meta(
+        targets=targets_list,
+        owner=owner,
+        project_name=project_name,
+        description=description,
+    )
+
+    logger.info(f"Using OBS project name = {project_name}")
+
+    project_url = core.makeurl(
+        _API_URL,
+        ["source", project_name, "_meta"],
+    )
+    metafile = core.metafile(project_url, ET.tostring(project_metadata))
+    metafile.sync()
+
+    package_names = list(package_config.packages.keys())
+
+    if len(package_names) != 1:
+        raise ValueError("Cannot handle multiple packages in package_config")
```

Comment:
Why not?

Reply:
osc does not support batch builds on same session. 

Might be OBS? 
@dcermak what do you think?

Reply:
OBS supports building multiple specs from the same package, but it would make matters more complex. Omitting this less used case allowed us to create a prototype. We can still implement this later on, if need be